wordpress-mobile/WordPress-iOS
# Adds the new domain purchase flow

Fixes #17150

Related [WordPressKit PR](https://github.com/wordpress-mobile/WordPressKit-iOS/pull/444)

This PR adds the new domain purchase flow to the Domains Dashboard

To test:
Pre-requisites
- It is recommended to use the store sandbox to avoid actual purchases while testing
- it is recommended to setup mock payment methods in the store sandbox to test more than one payment method (refer to the field guide for details)

Testing
- Build/run then go to My Site -> Domains
- Tap on "Get your domain"
- Either choose a domain from the predefined search or insert a custom search criteria, obtain a new list and choose a domain
- Tap on "Choose domain" and make sure that the domain purchase page opens in a web view (see screenshot below)
- Select a payment method of your choice and tap on Pay xxx with yyy (where xxx and yyy should report details about the price and the selected payment method)
- Make sure the purchase completes successfully. At this time, you will not see the native post purchase screen, and you should dismiss the web view manually

Known issues:
- The post purchase screen is not final. The web post purchase screen will be shown when the purchase completes
- The Domains dashboard is not final, nor it will update after the successful domain purchase
- The existing redeeming flow for paid plans is unchanged, and will not show the web view
- The cart created with the purchase is not deleted if the view is dismissed before completing the purchase, so it will persist in the user's cart on the same store used for the purchase (sandbox or live)

<p align=center>
<img width="264"  src="https://user-images.githubusercontent.com/34376330/133178973-664fbc94-1494-41e6-8ef6-a85c998931f1.png">
</p>

## Regression Notes
1. Potential unintended areas of impact
None: this adds a  new flow

2. What I did to test those areas of impact (or what existing automated tests I relied on)


3. What automated tests I added (or what prevented me from doing so)

PR submission checklist:
